samples/subscriptions.js

```diff
@@ -363,75 +307,66 @@ function synchronousPull(projectName, subscriptionName) {
   }
 
   // The subscriber pulls a specific number of messages.
-  client
-    .pull(request)
-    .then(responses => {
-      // The first element of `responses` is a PullResponse object.
-      const response = responses[0];
-
-      // Initialize `messages` with message ackId, message data and `false` as
-      // processing state. Then, start each message in a worker function.
-      response.receivedMessages.forEach(message => {
-        messages[message.ackId] = [message.message.data, false];
-        worker(message);
-      });
-
-      let numProcessed = 0;
-
-      // setInterval() gets run every 10s.
-      const interval = setInterval(function() {
-        // Every 10s, we do a check on the processing states of the messages.
-        Object.keys(messages).forEach(ackId => {
-          if (messages[ackId][1]) {
-            // If the processing state for a particular message is true,
-            // We will ack the message.
-            const ackRequest = {
-              subscription: formattedSubscription,
-              ackIds: [ackId],
-            };
-
-            client.acknowledge(ackRequest).catch(err => {
-              console.error(err);
-            });
-
-            console.log(`Acknowledged: "${messages[ackId][0]}".`);
-
-            // Increment numProcessed by 1.
-            numProcessed += 1;
-
-            // Remove this message from `messages`.
-            delete messages[ackId];
-          } else {
-            // If the processing state of a particular message remains false,
-            // we will modify its ack deadline.
-            const modifyAckRequest = {
-              subscription: formattedSubscription,
-              ackIds: [ackId],
-              ackDeadlineSeconds: ackDeadlineSeconds,
-            };
-
-            client.modifyAckDeadline(modifyAckRequest).catch(err => {
-              console.error(err);
-            });
-
-            console.log(
-              `Reset ack deadline for "${
-                messages[ackId][0]
-              }" for ${ackDeadlineSeconds}s.`
-            );
-          }
-
-          // If all messages have been processed, we clear out of the interval.
-          if (numProcessed === response.receivedMessages.length) {
-            clearInterval(interval);
-            console.log(`Done.`);
-          }
+  const [response] = await client.pull(request);
+  // Initialize `messages` with message ackId, message data and `false` as
+  // processing state. Then, start each message in a worker function.
+  response.receivedMessages.forEach(message => {
+    messages[message.ackId] = [message.message.data, false];
+    worker(message);
+  });
+
+  let numProcessed = 0;
+
+  // setInterval() gets run every 10s.
+  const interval = setInterval(function() {
+    // Every 10s, we do a check on the processing states of the messages.
+    Object.keys(messages).forEach(async ackId => {
```

Comment:
With the change to make this async, it likely becomes a `.map` and `await Promise.all`